faunadb-java-dsl/src/main/java/com/faunadb/client/types/Value.java

```diff
@@ -137,6 +137,20 @@ private Value() {
     return Decoder.decode(this, Types.arrayListOf(elementType));
   }
 
+  /**
+   * Attempts to coerce this value to a {@link List}.
+   * This object must be an instance of {@link ArrayV}.
+   *
+   * @param <T> the type of the elements on the collection
+   * @param elementType the type of the elements in the collection.
+   * @return a {@link Result} with the resulting collection.
+   * @see Decoder
+   * @see Types
+   */
+  public final <T> Result<List<T>> asListOf(Class<T> elementType) {
```

Comment:
Isn't a `List` a subtype of `Collection`? Why do you need a `List` specifically?

Reply:
well, in terms of usability a collection is a mess, we cannot index and get the size directly